src/sentry/models/identity.py

```diff
@@ -84,5 +90,21 @@ def reattach(cls, idp, external_id, user, defaults):
         """
         lookup = Q(external_id=external_id) | Q(user=user)
         Identity.objects.filter(lookup, idp=idp).delete()
-
-        return Identity.objects.create(idp=idp, user=user, external_id=external_id, **defaults)
+        logger.info(
+            "deleted-identity",
+            extra={"external_id": external_id, "idp_id": idp.id, "user_id": user.id},
+        )
+
+        identity_model = Identity.objects.create(
```

Comment:
Do you want to catch any exceptions this raises and log those? That might help give more information on why it could be failing.

Reply:
@markstory I actually don't think this step is failing but I suppose it wouldn't hurt to add

Reply:
@markstory Actually, I am pretty sure if this failed here then the pipeline would just fail and generate a Sentry error. I am not certain, but I think the problem is on the `IntegrityError` that we catch later. I don't think it's wise to add a bunch of try/except blocks that at a lower level if they will bubble up later anyways.